snapcraft/internal/sources.py

```diff
@@ -325,17 +326,26 @@ def provision(self, dst, clean_target=True, keep_zip=False):
 class Local(Base):
 
     def pull(self):
-        if os.path.islink(self.source_dir):
+        if os.path.islink(self.source_dir) or os.path.isfile(self.source_dir):
             os.remove(self.source_dir)
-        elif (os.path.isdir(self.source_dir) and
-              not os.listdir(self.source_dir)):
-            os.rmdir(self.source_dir)
-        elif os.path.exists(self.source_dir):
-            raise EnvironmentError('Cannot pull to target {!r}'.format(
-                self.source_dir))
+        elif os.path.isdir(self.source_dir):
+            shutil.rmtree(self.source_dir)
```

Comment:
El 01/07/16 a las 18:25, Kyle Fazzari escribió:
> In snapcraft/internal/sources.py
> <https://github.com/snapcore/snapcraft/pull/622#discussion_r69357549>:
> 
>>              os.remove(self.source_dir)
>> -        elif (os.path.isdir(self.source_dir) and
>> -              not os.listdir(self.source_dir)):
>> -            os.rmdir(self.source_dir)
>> -        elif os.path.exists(self.source_dir):
>> -            raise EnvironmentError('Cannot pull to target {!r}'.format(
>> -                self.source_dir))
>> +        elif os.path.isdir(self.source_dir):
>> +            shutil.rmtree(self.source_dir)
> 
>     We do need to track source changes now though ;-)
> 
> I'm not sure what you mean. Are you saying something needs to happen here?

Eventually, when we implement tracking of changes to sources that would
affect build results ;)